DBSCE/Overview.md

```diff
@@ -0,0 +1,221 @@
+
+From the deployment point of view there are two types of local key helpers: _well-known_(_private_) and _third party_(_public_)
+
+- _Public local key helper_ or _third party_: Can be accessed by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP through as defined in DBSC(E) protocol. A third party local key helper has a special deployment mechanism. We expect to loosely define the API and deployment mechanism in the upcoming specification, and will leave the specific implementation details to the vendors. However, we have provided examples of how this can look for a few platforms [below](#platform-examples).
+- _Private local key helper_ or _well known_ : Can be specific to an IdP. Typically owned by the IdP and will have a private protocol to communicate with the IdP. Comes with either OS or built into the browser. It assumed that well-known key helpers are **trusted and enabled by default** in a browser and/or a given IDP. A browser knows how to activate a well-known key helper.
+
+The Local Key Helper is responsible for:
+
+- Generation of the binding key and producing binding statements (see below)
+- Producing signatures with the binding key
+- Cleanup of the binding key and its artifacts (when the user clears the browser session or the key is unused for a long time).
+
+#### Platform Examples
+
+- Windows: Please refer to the Windows Local Key Helper [here](./KeyGeneration.md#local-key-helper-on-windows) for an example of a Local Key Helper.
+- MacOS: TBD
+- Android:TBD
+
+Note: Above are examples of Local Key Helpers that can be used for DBSC(E) key generation. Any platform is free to build a version of this, as long as the API is consistent with the protocol.
+
+### Attestation Service:
+
+A service that is responsible for verifying that the binding key is issued by the expected device and providing the attestation of the key to the IdP. The attestation service can be owned by the IdP or a third party. DBSC(E) relies on the attestation service to validate the binding statement and ensure that the binding key and the device key belong to the same device. We have added details on the specifics of the binding artifacts generated during the device registration process, and the validation of the binding statement in the [DBSC(E) Key Generation](#key-generation-specifics) section.
+
+### Device Registration (Pre-Session)
+
+Any enterprise user is expected to either use a device issued by their organization or register their personal device with the organization. The device registration is expected to be a once-in-a-lifetime operation, and the user is expected to perform this operation in a clean room environment.
+
+![DeviceRegistration](./DeviceRegistration.svg)
```

Comment:
I think we also need to describe the terminology of "binding and attestation keys"  before going into the design section.

Reply:
Yeah, I moved the definitions to a separate document to first layout the design. Key generation is incomplete as is. We have all platform specifics now, so I will restructure it.

Reply:
Added key specifics.